### PR TITLE
Fix for loop bug with Redis down and fix for uncaught errors in spawned worker's redis

### PR DESCRIPTION
**Pull request by fatso83**
This PR fixes the loop problems mentioned in #5, that is caused by a recursive synchronous function call, by passing the job task on to `nextTick`.

There was also a problem with the internal redis client in each spawned worker emitting errors that were not listened to anywhere. That killed our Node process when Redis disconnected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "redis-dist-job-queue",
-  "version": "0.1.0",
+  "version": "0.1.1",
```
Comment (andrewrk) [inline package.json]: I typically do the version bumping when I publish, no need to put it in the PR.

File: lib/index.js [hunk 2]
```diff
@@ -265,10 +265,12 @@ JobQueue.prototype.shutdown = function(callback) {
...
-    self.redisClient.quit(function(err, result) {
-      if (err) {
-        self.emit('error', err);
-      }
+    self.redisClient.quit(function(err) {
+        /* We want to kill this instance of the redis client, so at at this point we
+         * are not interested in what connection errors Redis comes up with.
+         */
+
+        self.redisClient.end();
```
Comment (andrewrk) [inline lib/index.js]: hmm, why wasn't this call done in the original code?

File: package.json (hunk 1 above)
Comment (fatso83) [inline package.json]: Sorry about that: want me to remove it before merge, or do you want to do it yourself?

File: lib/index.js (hunk 2 above)
Comment (fatso83) [inline lib/index.js]: If you by 'original code' mean the calling client code, then it is simply by your API design. The redis client is an internal property of your API. As a client I am simply supplying you with `{redisConfig : { port : ..., host : ..., pass }}`, not an actual client, so the control of the redis client is an internal detail of your implementation on which I cannot rely. You might rename it in later versions, or do something entirely different.

If you on the other hand asked for client to supply you with redis clients in the constructor, then we would have a direct handle to the redis client and we could configure that as we pleased. 

So in conclusion: as the redis client is your responsibility, then you also need to make that it does not cause problems for the clients when the client is using your API to shut down the job queue.

**Comment by fatso83** [comment]
Any problems with the code?

**Comment by fatso83** [comment]
Or do you perhaps suggest adding something to the API? This could also be fixed by letting the API accept a pre-configured redis instance.

**Comment by fatso83** [comment]
OK, giving up.

**Comment by andrewrk** [comment]
Hey I'm really sorry. Can I hand this module to you for maintenance and ownership? By that I mean transfer the repo to you and everything

**Comment by fatso83** [comment]
Sure thing!

**Comment by andrewrk** [comment]
Cool. what is your npm username?

**Comment by andrewrk** [comment]
> fatso83/node-redis-dist-job-queue already exists

it will also fail if you rename the repo. you have to completely delete your fork on github in order for me to initiate a transfer.

**Comment by fatso83** [comment]
done

**Comment by andrewrk** [comment]
transfer initiated. npm username?

**Comment by andrewrk** [comment]
nvm: https://www.npmjs.com/~fatso83

**Comment by fatso83** [comment]
Thanks!

**Comment by andrewrk** [comment]
No problem. Thanks for taking care of the project.
